### PR TITLE
editoast: electrical profile set use derive for retrive and create

### PR DESCRIPTION
**Pull request by Tristramg**

### CODE REVIEW

File: editoast/src/views/electrical_profiles.rs [hunk 1]
```diff
@@ -41,10 +41,13 @@ async fn get(
     db_pool: Data<DbPool>,
     electrical_profile_set: Path<i64>,
 ) -> Result<Json<ElectricalProfileSetData>> {
-    let electrical_profile_set = electrical_profile_set.into_inner();
+    let electrical_profile_set_id = electrical_profile_set.into_inner();
     block(move || {
         let mut conn = db_pool.get()?;
-        let ep_set = ElectricalProfileSet::retrieve(&mut conn, electrical_profile_set)?;
+        let ep_set = ElectricalProfileSet::retrieve_conn(&mut conn, electrical_profile_set_id)?
```
Comment (leovalais) [inline editoast/src/views/electrical_profiles.rs]: Here you can use `retrieve(db_pool, id).await?` in order to get rid of the `block`

File: editoast/src/views/electrical_profiles.rs [hunk 2]
```diff
@@ -77,41 +80,20 @@ async fn get_level_order(
     db_pool: Data<DbPool>,
     electrical_profile_set: Path<i64>,
 ) -> Result<Json<HashMap<String, Vec<String>>>> {
-    let electrical_profile_set = electrical_profile_set.into_inner();
+    let electrical_profile_set_id = electrical_profile_set.into_inner();
     block(move || {
         let mut conn = db_pool.get()?;
-        let ep_set = ElectricalProfileSet::retrieve(&mut conn, electrical_profile_set)?;
+        let ep_set = ElectricalProfileSet::retrieve_conn(&mut conn, electrical_profile_set_id)?
```
Comment (leovalais) [inline editoast/src/views/electrical_profiles.rs]: Likewise, retrieve(pool, id)

File: editoast/src/views/train_schedule/mod.rs [hunk 3]
```diff
@@ -527,10 +529,13 @@ async fn create_backend_request_payload(
         })?;
 
     let electrical_profile_set = match scenario.electrical_profile_set_id {
-        Some(Some(id)) => {
+        Some(Some(electrical_profile_set_id)) => {
             let mut conn = db_pool.get()?;
-            let eps = ElectricalProfileSet::retrieve(&mut conn, id)?;
-            eps.name
+            ElectricalProfileSet::retrieve_conn(&mut conn, electrical_profile_set_id)?
```
Comment (leovalais) [inline editoast/src/views/train_schedule/mod.rs]: Likewise, `retrive(pool, id)` (there should have been a block wrapping the conn here 👀)

**Review by leovalais — APPROVED** [review]
LGTM, a few suggestions:

**Comment by Tristramg** [comment]
All the suggestions have been applied, thank you!

**Review by leovalais — APPROVED** [review]

**Pull request merged**